### ISSUE TITLE
/wrkdirs/usr/ports/graphics/alizams/work/AlizaMS-1.9.3/mdcm/Source/Common/mdcmSwapper.h:92:16: error: read-only variable is not assignable

### ISSUE THREAD

**Issue by pkubaj** [AlizaMedicalImaging/AlizaMS#40]
Build fails on big-endian architectures:
```
In file included from /wrkdirs/usr/ports/graphics/alizams/work/AlizaMS-1.9.3/mdcm/Source/MediaStorageAndFileFormat/mdcmSegmentedPaletteColorLookupTable.cxx:23:
In file included from /wrkdirs/usr/ports/graphics/alizams/work/AlizaMS-1.9.3/mdcm/Source/DataStructureAndEncodingDefinition/mdcmDataSet.h:28:
In file included from /wrkdirs/usr/ports/graphics/alizams/work/AlizaMS-1.9.3/mdcm/Source/DataStructureAndEncodingDefinition/mdcmElement.h:31:
/wrkdirs/usr/ports/graphics/alizams/work/AlizaMS-1.9.3/mdcm/Source/Common/mdcmSwapper.h:92:16: error: read-only variable is not assignable
      array[i] = Swap<T>(array[i]);
      ~~~~~~~~ ^
/wrkdirs/usr/ports/graphics/alizams/work/AlizaMS-1.9.3/mdcm/Source/MediaStorageAndFileFormat/mdcmSegmentedPaletteColorLookupTable.cxx:233:18: note: in instantiation of function template specialization 'mdcm::SwapperNoOp::SwapArray<const unsigned short>' requested here
    SwapperNoOp::SwapArray(array16, length / 2);
                 ^
```
The test environment is FreeBSD 14.0 on powerpc64.

**Comment by pkubaj** [comment]
It works with the following patch:
```
--- /wrkdirs/usr/ports/graphics/alizams/work/AlizaMS-1.9.3/mdcm/Source/MediaStorageAndFileFormat/mdcmSegmentedPaletteColorLookupTable.cxx.orig  2023-09-27 13:26:01.759627000 +0000
+++ /wrkdirs/usr/ports/graphics/alizams/work/AlizaMS-1.9.3/mdcm/Source/MediaStorageAndFileFormat/mdcmSegmentedPaletteColorLookupTable.cxx       2023-09-27 13:25:52.992080000 +0000
@@ -225,8 +225,8 @@
     {
       copy[x] = array[x];
     }
-    const void *          varray = static_cast<void*>(copy);
-    const uint16_t *      array16 = static_cast<const uint16_t *>(varray);
+    void *          varray = static_cast<void*>(copy);
+    uint16_t *      array16 = static_cast<uint16_t *>(varray);
     std::vector<uint16_t> palette;
     unsigned int          num_entries = GetLUTLength(type);
     palette.reserve(num_entries);
```

**Comment by issakomi** [comment]
Thank you very much. Fixed in #41.

P.S. I have to admit that big-endian architectures have not been tested properly.
